pkg/actuators/machine/machine_scope.go

```diff
@@ -83,41 +88,67 @@ func newMachineScope(params machineScopeParams) (*machineScope, error) {
 	return mscp, nil
 }
 
-func (s *machineScope) getNutanixCredentials() {
-	credsSecretName := defaultNutanixCredsSecretName
-	if s.providerSpec.CredentialsSecret != nil {
-		credsSecretName = s.providerSpec.CredentialsSecret.Name
+func (s *machineScope) getNutanixCredentials() error {
+	// Get the PC endpoint/port from the Infrastructure CR
+	infra := &configv1.Infrastructure{}
+	infraKey := runtimeclient.ObjectKey{
+		Name: globalInfrastuctureName,
+	}
+	err := s.client.Get(s.Context, infraKey, infra)
+	if err != nil {
+		err1 := fmt.Errorf("Could not find the Infrastruture object %q: %v", infraKey.Name, err)
+		klog.Errorf("Machine %s: %v", err1.Error())
+		return err1
+	}
+
+	// Set the corresponding environment variable
+	pcEndpoint := infra.Spec.PlatformSpec.Nutanix.PrismCentral.Address
+	pcPort := infra.Spec.PlatformSpec.Nutanix.PrismCentral.Port
+	if len(pcEndpoint) == 0 {
+		return fmt.Errorf("The prismCentralEndpoint field is not set in the Infrastreucture CR")
+	}
+	os.Setenv(clientpkg.NutanixEndpointKey, pcEndpoint)
+	if pcPort < 1 || pcPort > 65535 {
+		return fmt.Errorf("The pcPort field is not set right in the Infrastreucture CR: %d", pcPort)
+	}
+	os.Setenv(clientpkg.NutanixPortKey, strconv.Itoa(int(pcPort)))
+
+	if s.providerSpec.CredentialsSecret == nil || len(s.providerSpec.CredentialsSecret.Name) == 0 {
+		return fmt.Errorf("The nutanix providerSpec credentialsSecret reference is not set.")
 	}
+	credsSecretName := s.providerSpec.CredentialsSecret.Name
 	credsSecret := &corev1.Secret{}
 	credsSecretKey := runtimeclient.ObjectKey{
 		Namespace: s.machine.Namespace,
 		Name:      credsSecretName,
 	}
-	err := s.client.Get(s.Context, credsSecretKey, credsSecret)
+	err = s.client.Get(s.Context, credsSecretKey, credsSecret)
 	if err != nil {
-		klog.Warningf("[Machine: %s] Could not find the credentials secret %s", s.machine.Name, credsSecretKey.Name)
-		return
+		err1 := fmt.Errorf("Could not find the local credentials secret %s: %v", credsSecretKey.Name, err)
+		klog.Errorf("Machine %s: %v", err1.Error())
```

Comment:
You have two fmt substitutions here but only one variable passed, this will cause an error

Reply:
Fixed.